cts/cli/regression.tools.exp

```diff
@@ -7899,46 +7899,58 @@ crm_shadow: Could not access shadow instance 'cts-cli': No such file or director
+    </errors>
+  </status>
 </pacemaker-result>
 =#=#=#= End test: Verify another file-specified invalid configuration, outputting as xml - Invalid configuration (78) =#=#=#=
 * Passed: crm_verify     - Verify another file-specified invalid configuration, outputting as xml
 =#=#=#= Begin test: Verbosely verify a file-specified invalid configuration, outputting as xml =#=#=#=
 unpack_config 	warning: Blind faith: not fencing unseen nodes
-(native_unpack@native.c:236) 	error: Resource test2:0 is of type systemd and therefore cannot be used as a promotable clone resource
-unpack_resources 	error: Ignoring <clone> resource 'test2-clone' because configuration is invalid
-crm_verify: Errors found during check: config not valid
 <pacemaker-result api-version="X" request="crm_verify_invalid_bz.xml --output-as=xml --verbose">
-  <status code="78" message="Invalid configuration"/>
+  <status code="78" message="Invalid configuration">
+    <errors>
+      <error>Resource test2:0 is of type systemd and therefore cannot be used as a promotable clone resource</error>
+      <error>Ignoring &lt;clone&gt; resource 'test2-clone' because configuration is invalid</error>
+      <error>crm_verify: Errors found during check: config not valid</error>
+    </errors>
+  </status>
 </pacemaker-result>
 =#=#=#= End test: Verbosely verify a file-specified invalid configuration, outputting as xml - Invalid configuration (78) =#=#=#=
 * Passed: crm_verify     - Verbosely verify a file-specified invalid configuration, outputting as xml
 =#=#=#= Begin test: Verbosely verify another file-specified invalid configuration, outputting as xml =#=#=#=
 (cluster_status@status.c:113) 	warning: Fencing and resource management disabled due to lack of quorum
-unpack_resources 	error: Resource start-up disabled since no STONITH resources have been defined
-unpack_resources 	error: Either configure some or disable STONITH with the stonith-enabled option
-unpack_resources 	error: NOTE: Clusters with shared data need STONITH to ensure data integrity
-(schedule_fencing_and_shutdowns@pcmk_scheduler.c:599) 	warning: Node pcmk-1 is unclean but cannot be fenced
-(schedule_fencing_and_shutdowns@pcmk_scheduler.c:599) 	warning: Node pcmk-2 is unclean but cannot be fenced
-crm_verify: Errors found during check: config not valid
 <pacemaker-result api-version="X" request="crm_verify_invalid_no_stonith.xml --output-as=xml --verbose">
-  <status code="78" message="Invalid configuration"/>
+  <status code="78" message="Invalid configuration">
+    <errors>
+      <error>Resource start-up disabled since no STONITH resources have been defined</error>
+      <error>Either configure some or disable STONITH with the stonith-enabled option</error>
+      <error>NOTE: Clusters with shared data need STONITH to ensure data integrity</error>
+      <error>Node pcmk-1 is unclean but cannot be fenced</error>
+      <error>Node pcmk-2 is unclean but cannot be fenced</error>
```

Comment:
I believe the text output is still the same, it's just the new XML output that's affected. It would be helpful to distinguish warnings there too, but it could be a significant project -- there's currently no warn() method for output objects.

Reply:
> I believe the text output is still the same, it's just the new XML output that's affected. 

Prefixes for text output are unfortunately missing too.

Reply:
I believe you always had to add -V to get details with prefixes.

Reply:
Ahhh right, we switched those to use the output object instead of the logging functions. Well at least they are split for warning/error already. We could add a warn() method to the output object, and the XML output could show it in a `<warning>` element. Text output is more a problem, since err() doesn't add a prefix, and we don't want to change that for all tools. Maybe we could create wrapper functions for crm_verify that add a prefix and then call err()/warn() (for use with `pcmk__set_config_error_handler()` etc.).

Reply:
Actually the fix should be pretty easy: https://projects.clusterlabs.org/T834